### PR TITLE
Add createAuthenticator mutation

### PR DESCRIPTION
**Pull request by tung2744**
ref DEV-1360

### CODE REVIEW

File: pkg/admin/graphql/authenticator_def.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		"password": &graphql.InputObjectFieldConfig{
+			Type:        graphql.String,
+			Description: "Password of the new authenticator.",
+		},
+	},
+})
+
+var authenticatorDef = graphql.NewInputObject(graphql.InputObjectConfig{
+	Name:        "IdentityDefinition",
+	Description: "Definition of an identity. This is a union object, exactly one of the available fields must be present.",
```
Comment (louischan-oursky) [inline pkg/admin/graphql/authenticator_def.go]: an authenticator, not an identity.

File: pkg/admin/graphql/authenticator_mutation.go [hunk 2]
```diff
@@ -78,3 +78,42 @@ var _ = registerMutationField(
...
+var createAuthenticatorInput = graphql.NewInputObject(graphql.InputObjectConfig{
+	Name: "CreateAuthenticatorInput",
+	Fields: graphql.InputObjectConfigFieldMap{
+		"userID": &graphql.InputObjectFieldConfig{
+			Type:        graphql.NewNonNull(graphql.ID),
+			Description: "Target user ID.",
+		},
+		"definition": &graphql.InputObjectFieldConfig{
+			Type:        graphql.NewNonNull(authenticatorDef),
+			Description: "Definition of the new identity.",
```
Comment (louischan-oursky) [inline pkg/admin/graphql/authenticator_mutation.go:92]: the new authenticator.

File: portal/src/graphql/adminapi/schema.graphql [hunk 3]
```diff
@@ -1,1675 +0,0 @@
-""""""
```
Comment (louischan-oursky) [inline portal/src/graphql/adminapi/schema.graphql]: Do not delete this file in this commit.

File: portal/src/graphql/adminapi/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,1734 @@
...
+input AuthenticatorDefinition {
+  """OOB OTP Email authenticator definition."""
+  oobOtpEmail: AuthenticatorDefinitionOOBOTPEmail
+
+  """OOB OTP SMS authenticator definition."""
+  oobOtpSMS: AuthenticatorDefinitionOOBOTPSMS
+
+  """Password authenticator definition."""
+  password: AuthenticatorDefinitionPassword
+}
```
Comment (louischan-oursky) [inline portal/src/graphql/adminapi/schema.graphql:507]: I prefer the following shape. What do you think?

```graphql
input AuthenticatorDefinition {
  kind: AuthenticatorKind!
  """We will check type, and determine which field we will then look at."""
  type: AuthenticatorType!
  oobOtpEmail: AuthenticatorDefinitionOOBOTPEmail
  oobOtpSMS: AuthenticatorDefinitionOOBOTPSMS
  password: AuthenticatorDefinitionPassword
}

input AuthenticatorDefinitionOOBOTPEmail {
  email: String!
}

input AuthenticatorDefinitionOOBOTPSMS {
  phone: String!
}

input AuthenticatorDefinitionPassword {
  password: String!
}
```

**Comment by tung2744** [comment]
@louischan-oursky Updated, thanks!

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**